python/deptry/dependency_getter/builder.py

```diff
@@ -45,21 +45,89 @@ def build(self) -> DependencyGetter:
+                    self.package_module_name_map,
+                    self.pep621_dev_dependency_groups,
                 )
 
         check, requirements_files = self._project_uses_requirements_files()
         if check:
             return RequirementsTxtDependencyGetter(
-                self.config, self.package_module_name_map, requirements_files, self.requirements_files_dev
+                self.config,
+                self.package_module_name_map,
+                requirements_files,
+                self.requirements_files_dev,
             )
 
         raise DependencySpecificationNotFoundError(self.requirements_files)
 
+    def _project_uses_setuptools(self, pyproject_toml: dict[str, Any]) -> bool:
+        try:
+            if pyproject_toml["build-system"]["build-backend"] == "setuptools.build_meta":
+                logging.debug(
+                    "pyproject.toml has the entry"
+                    " build-system.build-backend == 'setuptools.build_meta'"
+                    ", so setuptools is used to specify the project's dependencies."
+                )
+                return True
+            else:
+                logging.debug(
+                    "pyproject.toml does not have"
+                    " build-system.build-backend == 'setuptools.build_meta'"
+                    ", so setuptools is not used to specify the project's dependencies."
+                )
+                return False
+        except KeyError:
+            logging.debug(
+                "pyproject.toml does not contain a build-system.build-backend entry, "
+                "so setuptools is not used to specify the project's dependencies."
+            )
+            return False
+
+    def _project_uses_dynamic_dependencies(self, pyproject_toml: dict[str, Any]) -> bool:
+        try:
+            if "dependencies" not in pyproject_toml["project"]["dynamic"]:
+                logging.debug(
+                    "pyproject.toml does not have"
+                    " dependencies listed as dynamic metadata,"
+                    " so dynamic dependencies are not used."
+                )
+                return False
+            else:
+                pyproject_toml["tool"]["setuptools"]["dynamic"]["dependencies"]["file"]
+                logging.debug(
+                    "pyproject.toml has dependencies listed as dynamic metadata"
+                    " and contains a populated tool.setuptools.dynamic.dependencies.file"
+                    " entry, so dynamic dependencies are used."
+                )
+                return True
+        except KeyError:
+            logging.debug(
+                "pyproject.toml either does not contain a project.dynamic entry or "
+                "tool.setuptools.dynamic.dependencies.file entry, so dynamic "
+                "dependencies are not used."
+            )
+            return False
+
+    def _project_dynamic_requirements_file(self, pyproject_toml: dict[str, Any]) -> tuple[str, ...]:
```

Comment:
Maybe we can remove this method and make `_project_uses_dynamic_dependencies` return a `tuple[bool, tuple[str, ...]]`, like https://github.com/fpgmaas/deptry/blob/5756281f54a1e3395300b78251a50b9563c770fd/python/deptry/dependency_getter/builder.py#L119?